pkg/controller/kubebuilderx/plan_builder.go

```diff
@@ -172,9 +175,38 @@ func buildOrderedVertices(ctx context.Context, currentTree *ObjectTree, desiredT
 		for name := range updateSet {
 			oldObj := oldSnapshot[name]
 			newObj := newSnapshot[name]
-			if !reflect.DeepEqual(oldObj, newObj) {
-				v := model.NewObjectVertex(oldObj, newObj, model.ActionUpdatePtr(), inDataContext4G())
-				findAndAppend(v)
+			if !equality.Semantic.DeepEqual(oldObj, newObj) {
+				// if object is pod and using InPlacePodVerticalScaling, it may need to use a /resize subresource. Update the sub resource if needed.
+				useResizeSubResource := false
+				if oldPod, ok := oldObj.(*corev1.Pod); ok {
+					newPod := newObj.(*corev1.Pod)
+					equal := true
+					for i, c := range oldPod.Spec.Containers {
+						newC := newPod.Spec.Containers[i]
```

Comment:
Check if the length of the `Containers` slice is the same to avoid out-of-bound errors. Although this is not allowed by k8s, our application code might make mistakes, and proper code defense is necessary.